### PR TITLE
Improve the app submission and review process

### PR DESCRIPTION
**Pull request by zeke**
## Why?

The existing setup of using a single `_apps.yml` for collecting all app data is flawed: Each new app is added to the end of the list. This means that merging any app PR generates a merge conflict for all existing PRs. This was not an issue when we only got one app submission every few days, but now we're getting more of them. We currently have 23 open PRs, and we don't want to have to tell/teach 22 users to rebase.

We also waste time reviewing PRs and having back-and-forth discussions with contributors about  small problems like empty metadata, e.g. `license: ''`.

## How?

This PR moves the app data to a new repository, https://github.com/electron/electron-apps which has a comprehensive test suite to make sure that all the submitted apps meet the requirements:

- `name` is required
- `name` is unique
- `description` is required
- `website` is required and must be a URL
- `website` is optional, but must be a URL if provided
- no properties can have empty values
- icon must be a png
- icon must be square
- [etc](https://github.com/electron/electron-apps/blob/5475ff35411c04975ac9775cc256a3ea75ebce45/test.js)...

This allows us to let Travis CI do most of the review work for us.

Rather than requiring numerous publishes to the npm registry, the new repo is consumed by the Electron website as a GitHub-hosted package: `github:electron/electron-apps`

## Wizard 🔮

I added an interactive wizard for submitting apps. It looks like this:

screen shot 2016-12-09 at 9 35 44 am

This wizard is documented in the [electron-apps README](https://github.com/electron/electron-apps#adding-your-app), as is a [manual process](https://github.com/electron/electron-apps#adding-your-app-by-hand-) for folks who are submitting PRs using the website, or who don't want to use the wizard. This new documentation is now also linked to from the website's `CONTRIBUTING.md` file. 

## Notes

- To update the app data on the website, run `npm run apps`
- Apps are now in alphabetical order, not chronological order
- Removed the `featuredOnHomepage` flag from app metadata. Now it's all in a separate list in `_data/featured_apps.json`

This is ready for 👀 , @electron/maintainers

### CODE REVIEW

File: spec/featured-spec.js [hunk 1]
```diff
@@ -4,10 +4,6 @@ var yaml = require('yamljs')
 
 test('Limit featured homepage apps to 24', function (t) {
   t.plan(1)
-
-  var apps = yaml.load(path.join(__dirname, '..', '_data', 'apps.yml'))
-  var featuredApps = apps.filter(function (app) {
-    return app.featuredOnHomepage
-  })
+  var featuredApps = require(path.join(__dirname, '../_data/featured_apps.json'))
```
Comment (50Wliu) [inline spec/featured-spec.js]: This should still be split up to work on Windows

Comment (zeke) [inline spec/featured-spec.js]: Oh really!? You mean the `'../_data/featured_apps` part? I thought `path.join` took care of that. I guess that means I've been writing a lot of windows-unfriendly path code. 🙈

Comment (zeke) [inline spec/featured-spec.js]: Updated. Thanks for noticing, @50Wliu!

Comment (50Wliu) [inline spec/featured-spec.js]: Hmm.  Looks like I was actually wrong...seems like I remembered this not working, but I just tested and path.join fixes the slashes now.  Sorry!

**Comment by zeke** [comment]
Update: I closed all the extant app-submission PRs on this repo and moved them over: https://github.com/electron/electron-apps/pulls?utf8=%E2%9C%93&q=is%3Apr

Thanks to @kevinsawicki for the push. 🚀

**Comment by zeke** [comment]
@kevinsawicki pointed out today that if we ship this PR, newly added apps will not show up on the site automatically. We'd have to run `npm run apps` then commit and push.

To work around this, I've added a [release script](https://github.com/electron/electron.atom.io/blob/15bf62ae3a8a48f2ef3f43825478c8ae10e05dfd/script/release) that we can run every ten minutes in a scheduled task on Heroku.

In addition to new docs, this script will also take care of auto-updating the docs as new versions of Electron are released.

**Pull request merged**